src/ir/term/mod.rs

```diff
@@ -721,6 +723,104 @@ impl Array {
         self.check_idx(idx);
         self.map.get(idx).unwrap_or(&*self.default).clone()
     }
+
+    /// Iter
+    pub fn into_iter(&self) -> std::collections::btree_map::IntoIter<Value, Value> {
+        self.map.clone().into_iter()
+    }
+}
+
+/// Merge two Array Iterators
+pub struct ArrayMerge {
+    left: Peekable<Box<std::collections::btree_map::IntoIter<Value, Value>>>,
+    right: Peekable<Box<std::collections::btree_map::IntoIter<Value, Value>>>,
+    left_dfl: Value,
+    right_dfl: Value,
+}
+
+impl ArrayMerge {
+    /// Create a new [ArrayMerge] from two [Array]s
+    pub fn new(a: Array, b: Array) -> Self {
+        if a.size != b.size {
+            panic!("IR Arrays have different lengths: {}, {}", a.size, b.size);
+        }
+        if a.key_sort != b.key_sort {
+            panic!(
+                "IR Arrays have different key sorts: {}, {}",
+                a.key_sort, b.key_sort
+            );
+        }
+        if a.default.sort() != b.default.sort() {
+            panic!(
+                "IR Arrays default values have different key sorts: {}, {}",
+                a.default.sort(),
+                b.default.sort()
+            );
+        }
```

Comment:
`assert!()` is more compact than if-`panic!` and means exactly the same thing. You can even [add error messages](https://doc.rust-lang.org/std/macro.assert.html#custom-messages) in the same way.